hardware/client.go

```diff
@@ -13,22 +13,26 @@ import (
 	tinkClient "github.com/tinkerbell/tink/client"
 	tpkg "github.com/tinkerbell/tink/pkg"
 	tink "github.com/tinkerbell/tink/protos/hardware"
-	"google.golang.org/grpc"
 )
 
 var (
 	dataModelVersion = env.Get("DATA_MODEL_VERSION")
 	facility         = flag.String("facility", env.Get("HEGEL_FACILITY", "onprem"), "The facility we are running in (mostly to connect to cacher)")
 )
 
-// Client acts as the messenger between Hegel and Cacher/Tink.
+// Client defines the behaviors for interacting with hardware data providers.
 type Client interface {
-	All(ctx context.Context, opts ...grpc.CallOption) (AllClient, error)
-	ByIP(ctx context.Context, id string, opts ...grpc.CallOption) (Hardware, error)
-	Watch(ctx context.Context, id string, opts ...grpc.CallOption) (Watcher, error)
-}
+	// IsHealth reports whether the client is connected and can retrieve hardware data from the data provider. Clients
+	// should report nil if healthy.
+	IsHealthy(ctx context.Context) error
```

Comment:
I'd expect Is* to return bool not error. If it were a bool we wouldn't need the comment explaining how to signal health either.

Reply:
I had it that way originally and thought the semantics for 'unhealthy' might need inspecting by consumers to decide if they care or not. 

I'll change to bool as I prefer the simplicity of it.

Reply:
Adjusted. Feel free to unresolve if you have another thought.